### ISSUE TITLE
sample.py doesn't report HTTP 400, says "no results" instead.

### ISSUE THREAD

**Issue by mittonk** [Yelp/yelp-fusion#321]
## Overview
- **Client ID:** ssWONR4BhYiARowH3vsVEA
- **Issue type:**  Bug
- **Summary:** sample.py doesn't report HTTP 400, says "no results" instead.
- **Platform:** Desktop, Python.

## Description
See #320 for resulting confusion.

Looks like the code is expecting to get a 400 raised as an HTTPError, but it's not --- so 
```
businesses = response.get('businesses')
```
just gets [].

### Endpoint
Search for sure.  May hit Business as well.

### Parameters or Sample Request
In sample.py, set
`SEARCH_LIMIT=9999`.  Run sample.py.

### Response
```
{
	"error": {
		"code": "VALIDATION_ERROR",
		"description": "The supplied limit: 1000 is invalid, the maximum is 50."
	}
}
```
However, this gets reported as:
`No businesses for restaurants in San Francisco`

**Comment by coolboiime** [comment]
I don’t think this is a bug. It is the limit imposed by Yelp. 50 requests at a time, 1000 requests overall through offsetting the results.
